harness/determined/pytorch/dsat/_dsat_search_method.py

```diff
@@ -1104,6 +1105,224 @@ def get_random_trial(self) -> DSATTrial:
         return random_trial
 
 
+@dataclass
+class ASHADSATSearchData:
+    lo: int
+    hi: int
+    curr_rung: int
```

Comment:
Mostly a note for myself, we will have to make this inherit from the `DSATSearchData` https://github.com/determined-ai/determined/pull/6846/files